File: ios/EmojiSheetTheme.swift
import UIKit

struct EmojiSheetTheme {
    let backgroundColor: UIColor
    let searchBarBackgroundColor: UIColor
    let textColor: UIColor
    let textSecondaryColor: UIColor
    let dividerColor: UIColor
    let accentColor: UIColor
    let categoryIconColor: UIColor
    let categoryActiveIconColor: UIColor
    let categoryActiveBackgroundColor: UIColor
    let handleColor: UIColor
    let categoryBarBackgroundColor: UIColor
    let searchTextColor: UIColor?
    let placeholderTextColor: UIColor?
    let selectionColor: UIColor?

    init(
        backgroundColor: UIColor,
        searchBarBackgroundColor: UIColor,
        textColor: UIColor,
        textSecondaryColor: UIColor,
        dividerColor: UIColor,
        accentColor: UIColor,
        categoryIconColor: UIColor,
        categoryActiveIconColor: UIColor? = nil,
        categoryActiveBackgroundColor: UIColor? = nil,
        handleColor: UIColor? = nil,
        categoryBarBackgroundColor: UIColor? = nil,
        searchTextColor: UIColor? = nil,
        placeholderTextColor: UIColor? = nil,
        selectionColor: UIColor? = nil
    ) {
        self.backgroundColor = backgroundColor
        self.searchBarBackgroundColor = searchBarBackgroundColor
        self.textColor = textColor
        self.textSecondaryColor = textSecondaryColor
        self.dividerColor = dividerColor
        self.accentColor = accentColor
        self.categoryIconColor = categoryIconColor
        self.categoryActiveIconColor = categoryActiveIconColor ?? accentColor
        self.categoryActiveBackgroundColor = categoryActiveBackgroundColor ?? dividerColor
        self.handleColor = handleColor ?? UIColor(white: 0.72, alpha: 1)
        self.categoryBarBackgroundColor = categoryBarBackgroundColor ?? backgroundColor
        self.searchTextColor = searchTextColor
        self.placeholderTextColor = placeholderTextColor
        self.selectionColor = selectionColor
    }

    static let dark = EmojiSheetTheme(
        backgroundColor: UIColor(red: 0.1, green: 0.1, blue: 0.18, alpha: 1),
        searchBarBackgroundColor: UIColor(red: 0.16, green: 0.18, blue: 0.22, alpha: 1),
        textColor: .white,
        textSecondaryColor: UIColor(white: 1, alpha: 0.6),
        dividerColor: UIColor(white: 1, alpha: 0.15),
        accentColor: UIColor(red: 0.918, green: 0.271, blue: 0.471, alpha: 1),
        categoryIconColor: UIColor(white: 1, alpha: 0.5),
        categoryActiveIconColor: UIColor(red: 0.918, green: 0.271, blue: 0.471, alpha: 1),
        categoryActiveBackgroundColor: UIColor(white: 1, alpha: 0.15),
        handleColor: UIColor(white: 0.5, alpha: 1),
        categoryBarBackgroundColor: UIColor(red: 0.1, green: 0.1, blue: 0.18, alpha: 1)
    )

    static let light = EmojiSheetTheme(
        backgroundColor: .white,
        searchBarBackgroundColor: UIColor(red: 0.94, green: 0.94, blue: 0.94, alpha: 1),
        textColor: .black,
        textSecondaryColor: UIColor(white: 0, alpha: 0.5),
        dividerColor: UIColor(red: 0.878, green: 0.878, blue: 0.878, alpha: 1),
        accentColor: UIColor(red: 0.918, green: 0.271, blue: 0.471, alpha: 1),
        categoryIconColor: UIColor(white: 0, alpha: 0.4),
        categoryActiveIconColor: UIColor(red: 0.918, green: 0.271, blue: 0.471, alpha: 1),
        categoryActiveBackgroundColor: UIColor(red: 0.878, green: 0.878, blue: 0.878, alpha: 1),
        handleColor: UIColor(white: 0.72, alpha: 1),
        categoryBarBackgroundColor: .white
    )
}

File: android/src/main/java/expo/community/modules/emojisheet/EmojiSheetTheme.kt
package expo.community.modules.emojisheet

data class EmojiSheetTheme(
    val backgroundColor: Int,
    val searchBarBackgroundColor: Int,
    val textColor: Int,
    val textSecondaryColor: Int,
    val dividerColor: Int,
    val accentColor: Int,
    val categoryIconColor: Int,
    val categoryActiveIconColor: Int,
    val categoryActiveBackgroundColor: Int,
    val handleColor: Int,
    val categoryBarBackgroundColor: Int,
    val searchTextColor: Int? = null,
    val placeholderTextColor: Int? = null,
    val selectionColor: Int? = null
) {
    companion object {
        val dark = create(
            backgroundColor = 0xFF1A1A2E.toInt(),
            searchBarBackgroundColor = 0xFF2A2F39.toInt(),
            textColor = 0xFFFFFFFF.toInt(),
            textSecondaryColor = 0x99FFFFFF.toInt(),
            dividerColor = 0x26FFFFFF.toInt(),
            accentColor = 0xFFEA4578.toInt(),
            categoryIconColor = 0x80FFFFFF.toInt(),
            handleColor = 0x66FFFFFF.toInt()
        )

        val light = create(
            backgroundColor = 0xFFFFFFFF.toInt(),
            searchBarBackgroundColor = 0xFFF0F0F0.toInt(),
            textColor = 0xFF000000.toInt(),
            textSecondaryColor = 0x80000000.toInt(),
            dividerColor = 0xFFE0E0E0.toInt(),
            accentColor = 0xFFEA4578.toInt(),
            categoryIconColor = 0x66000000.toInt(),
            handleColor = 0x33000000.toInt()
        )

        fun fromName(name: String): EmojiSheetTheme =
            if (name == "dark") dark else light

        fun create(
            backgroundColor: Int,
            searchBarBackgroundColor: Int,
            textColor: Int,
            textSecondaryColor: Int,
            dividerColor: Int,
            accentColor: Int,
            categoryIconColor: Int,
            categoryActiveIconColor: Int = accentColor,
            categoryActiveBackgroundColor: Int = dividerColor,
            handleColor: Int = 0x33000000,
            categoryBarBackgroundColor: Int = backgroundColor,
            searchTextColor: Int? = null,
            placeholderTextColor: Int? = null,
            selectionColor: Int? = null
        ): EmojiSheetTheme = EmojiSheetTheme(
            backgroundColor = backgroundColor,
            searchBarBackgroundColor = searchBarBackgroundColor,
            textColor = textColor,
            textSecondaryColor = textSecondaryColor,
            dividerColor = dividerColor,
            accentColor = accentColor,
            categoryIconColor = categoryIconColor,
            categoryActiveIconColor = categoryActiveIconColor,
            categoryActiveBackgroundColor = categoryActiveBackgroundColor,
            handleColor = handleColor,
            categoryBarBackgroundColor = categoryBarBackgroundColor,
            searchTextColor = searchTextColor,
            placeholderTextColor = placeholderTextColor,
            selectionColor = selectionColor
        )
    }
}
